### PR TITLE
Eclipse cheatsheet

### PR DESCRIPTION
**Pull request by kishoresmeda**
#####  What does your Instant Answer do?

Display keyboard shortcuts for Eclipse IDE

#####  What is the data source for your Instant Answer? (Provide a link if possible)

http://www.shortcutworld.com/shortcuts.php?l=en&p=win&application=Eclipse

#####  Why did you choose this data source?

Nicely categorised.

#####  Are there any other alternative (better) data sources?

Didn't come across one.

#####  Which communities will this Instant Answer be especially useful for? (gamers, book lovers, etc)

Programmers, productivity enthusiasts, geeks, hackers etc.,

##### Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?

Yes. https://duck.co/ideas/idea/6042/keyboard-shortcuts-cheatsheet-for-eclipse-ide

#####  Are you having any problems? Do you need our help with anything?

No blockers so far. 

#####  Where did you hear about DuckDuckHack? (For first time contributors)

Meetup.com. Attended Duck Duck Hack here at Bangalore, India.

##### What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)

ddg

-----
IA Page: https://duck.co/ia/view/eclipse_cheat_sheet

### CODE REVIEW

**Comment by mintsoft** [comment]
@kishoresmeda thanks for the PR; can you fill out the PR template and paste it into the description?

**Comment by kishoresmeda** [comment]
@mintsoft  Need your help with PR template. It's my first time contributing to OSS or using git/github. Please assist. What's a PR template, where do i find it and where do i find the description where i can copy the filled in PR template. Sorry for too many questions!!

**Comment by altern8tif** [comment]
@kishoresmeda Edit the description on the your [comment](https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1271#issue-94485529) above using the template from this [file](https://github.com/duckduckgo/zeroclickinfo-goodies/blob/master/pull_request_template_cheat_sheet.md).

You can reference other cheatsheet pull requests to see an example.

**Comment by moollaza** [comment]
@kishoresmeda thanks for updating with the PR template (I've re-jigged your comments so it's a little easier for everyone to read)

File: share/goodie/cheat_sheets/eclipse.json [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+        "Refactoring",
+        "Run and Debug",
+        "The Rest"
+    ],
+
+    
+    "sections": {
+        "Manage Files and Projects": [
+            {
+                "key": "[Ctrl] + [N]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: Please remove the `+` in all these. We've actually updated the syntax a week ago or so in the documentation, but you may not have noticed ;)

```js
//bad
[Ctrl] + [N]

//good
[Ctrl] [N]
```

File: share/goodie/cheat_sheets/eclipse.json [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+            }
+            
+        ],        
+        "Editor Window": [
+            {
+                "key": "F12",
+                "val": "Jump to Editor Window"               
+            }, 
+            {
+                "key": "[Ctrl] + [Page Down] / [Ctrl] + [Page Up]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: Please see the documentation regarding alternatives.

In this case it should be:

```js
[Ctrl] ([Page Up]/[Page Down])"
```

File: share/goodie/cheat_sheets/eclipse.json [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+            {
+                "key": "[Alt] + [-]",
+                "val": "Open Editor Window Option menu"               
+            }, 
+            {
+                "key": "[Ctrl] + [F10]",
+                "val": "Show view menu "               
+            }, 
+            {
+                "key": "[Ctrl] + [F10], then [n]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: You can drop the `then` here

File: share/goodie/cheat_sheets/eclipse.json [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+                "val": "Show or hide line numbers"               
+            }, 
+            {
+                "key": "[Ctrl] + [Shift] +Q",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: Whoops, make sure that `Q` is wrapped: `[Q]` (also. don't forget to remove the `+`'s)

**Comment by moollaza** [comment]
ping! @kishoresmeda are you interested in finishing this up?

**Comment by kishoresmeda** [comment]
@moollaza Thanks for the ping! I'll finish this now.

**Comment by kishoresmeda** [comment]
@moollaza I've updated the code. Please take a look and let me know if any more changes are required.

File: share/goodie/cheat_sheets/eclipse.json [hunk 5]
```diff
@@ -0,0 +1,465 @@
+{
+   
+    "id": "eclipse_cheat_sheet", 
+    "name": "Eclipse",
+    "description": "Eclipse IDE", 
+    "metadata": { 
+        "sourceName": "Shortcut World",
+        "sourceUrl": "http://www.shortcutworld.com/en/win/Eclipse.html",
+        "primary_example_queries ": ["eclipse shortcuts", "eclipse ide keyboard shortcuts", "eclipse keyboard shortcuts"],
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: Please remove these. Only `sourceName` and `sourceUrl` are necessary. The IA Page is where some of the remaining info belongs ;)

File: share/goodie/cheat_sheets/eclipse.json [hunk 6]
```diff
@@ -0,0 +1,465 @@
...
+            {
+                "key": "[Ctrl] ([Home] / [End])",
+                "val": "Jump to beginning / jump to end of source"
+            },
+            {
+                "key": "[Ctrl] ([→] / [←])",
+                "val": "Jump one word to the left / one word to the right"               
+            }, 
+            {
+                "key": "[Ctrl-Shift] ([↓] / [↑])",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: Should be `[Ctrl] [Shift] ...`

File: share/goodie/cheat_sheets/eclipse.json [hunk 7]
```diff
@@ -0,0 +1,465 @@
...
+            {
+                "key": "[Shift] [Enter]",
+                "val": "Enter line below current line"               
+            }, 
+            {
+                "key": "[Shift] [Ctrl] [Enter]",
+                "val": "Enter line above current line"               
+            },
+            {
+                "key": "[Insert]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: Should be `Insert` no need for the `[]`

File: share/goodie/cheat_sheets/eclipse.json [hunk 8]
```diff
@@ -0,0 +1,465 @@
...
+                "val": "Step out"               
+            }, 
+            {
+                "key": "F8",
+                "val": "Skip to next Breakpoint"               
+            }
+        ],  
+        "The Rest": [
+            {
+                "key": "[Ctrl] ([F7] / [Shift] [F7])",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: Now that I see how these are rendering, I think we should break these into two separate key-value pairs:

```
{
    "key": "[Ctrl] [F7]".
    "val": "Move to next view (panel)"
},
{

    "key": "[Ctrl] [Shift] [F7]".
    "val": "Move to previous view (panel)"
}
```

File: share/goodie/cheat_sheets/eclipse.json [hunk 9]
```diff
@@ -0,0 +1,465 @@
...
+                "val": "Switch forward / backward between views (panels). Useful for switching back and forth between Package Explorer and Editor."               
+            }, 
+            {
+                "key": "[Ctrl] ([F8] / [Shift] [F8])",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: Let's break up all these kinds of key's that have a `[Shift]` in them into two KV pairs

File: share/goodie/cheat_sheets/eclipse.json [hunk 10]
```diff
@@ -0,0 +1,465 @@
...
+            }, 
+            {
+                "key": "[Ctrl] [Shift], [Q]",
+                "val": "Jump to Editor Window"               
+            }            
+            
+        ],  
+        "Navigate in Editor": [
+            {
+                "key": "[Home] / [End]",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: This should be 2 separate entries

File: share/goodie/cheat_sheets/eclipse.json [hunk 11]
```diff
@@ -0,0 +1,465 @@
...
+                "val": "Jump to beginning / jump to end of indention. Press home twice to jump to beginning of line"               
+            }, 
+            {
+                "key": "[Ctrl] ([Home] / [End])",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: This should be 2 separate entries as well

File: share/goodie/cheat_sheets/eclipse.json [hunk 12]
```diff
@@ -0,0 +1,465 @@
...
+            {
+                "key": "[Ctrl] [M]",
+                "val": "Maximize or un-maximize current Editor Window (also works for other Windows)"               
+            }, 
+            {
+                "key": "[Ctrl] [E]",
+                "val": "Show list of open Editors. Use arrow keys and enter to switch"               
+            }, 
+            {
+                "key": "[Ctrl] ([F6] / [Shift] [F6])",
```
Comment (moollaza) [inline share/goodie/cheat_sheets/eclipse.json]: Let's split this into 2 KV pairs

**Comment by moollaza** [comment]
@kishoresmeda almost there! After these edits it should be good for merge :+1:

**Comment by kishoresmeda** [comment]
Hi @moollaza!! Finished all the changes. See if it looks good.

**Comment by moollaza** [comment]
LGTM :+1:

**Pull request merged**